### DISCUSSION TITLE
Newb trying to get output-map working with external template

### DISCUSSION THREAD

**Discussion by dgarlitt** [hairyhenderson/gomplate#1676] [Q&A] [answered]
I was following the documentation [here](https://docs.gomplate.ca/usage/#referencing-complex-output-map-template-files) to learn how to use `--output-map` with external templates. I created the `out.t` file as indicated:

_out.t_
```go
{{- /* .in may contain a directory name - we want to preserve that */ -}}
{{ $f := filepath.Base .in -}}
out/{{ .in | strings.ReplaceAll $f (index .filemap $f) }}.out
```

as well as the filemap.json file:

_filemap.json_
```json
{ "eins.txt": "uno", "deux.txt": "dos" }
```

and stored my templates in the `in` directory. However, when I run the command in the example:

```shell
gomplate -t out=out.t -c filemap.json --input-dir=in --output-map='{{ template "out" }}'
``` 

I get the following error:

```shell
ERR  error="failed to gather templates for rendering: failed to render outputMap with ctx &map[ctx:0xc000a4e1b0 filemap:map[deux.txt:dos eins.txt:uno] in:deux.txt] and inPath deux.txt: failed to render template <OutputMap>: template: out:2:7: executing \"out\" at <.in>: nil data; no entry for key \"in\""
```

I am sure that I have a basic misunderstanding. Can someone please help point me in the right direction?

**Comment by hairyhenderson** [comment]
Hi @dgarlitt, thanks for reporting this - it looks like this is at least a bug in the documentation, and likely also a bug with how nested templates interact with output maps.

I tested this with various gomplate versions all the way back to v3.4.0 (when the output-map feature was added), and I get the same error too.

**Reply by hairyhenderson** [reply]
I've opened https://github.com/hairyhenderson/gomplate/issues/1686 to fix this

**Comment by hairyhenderson** [comment, marked as answer]
Many thanks to @thesuperzapper, [who pointed out](https://github.com/hairyhenderson/gomplate/issues/1686#issuecomment-1558033881) that the doc bug is that the context (`.`) needs to be passed to the `template` function for the example to work:

So, the correct command is:

```console
gomplate -t out=out.t -c filemap.json --input-dir=in --output-map='{{ template "out" . }}'
```

The docs are fixed in https://github.com/hairyhenderson/gomplate/pull/1745
